### PR TITLE
messaging: rewrite apt and motd messages

### PR DESCRIPTION
**Pull request by orndorffgrant**
## Test Steps

Use new integration tests

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

**Review by lucasmoura — COMMENTED** [review]
@orndorffgrant I think we should also update the `apt_messages` doc here

File: features/apt_messages.feature [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+        Building dependency tree...
+        Reading state information...
+        Calculating upgrade...
+        The following security updates require Ubuntu Pro with 'esm-infra' enabled:
+          .*
+        Learn more about Ubuntu Pro for 16\.04 at https:\/\/ubuntu\.com\/16-04
+        0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded\.
+        """
+        When I attach `contract_token` with sudo
+        When I run shell command `apt-get upgrade || true` with sudo
```
Comment (lucasmoura) [inline features/apt_messages.feature:122]: Why do we need the `|| true` here ?

File: features/staging_commands.feature [hunk 2]
```diff
@@ -30,53 +30,6 @@ Feature: Enable command behaviour when attached to an Ubuntu Pro staging subscri
         \s*\*\*\* .* 500
         \s*500 https://esm.staging.ubuntu.com/apps/ubuntu <release>-apps-security/main amd64 Packages
         """
-        When I run `mkdir -p /var/lib/ubuntu-advantage/messages` with sudo
```
Comment (lucasmoura) [inline features/staging_commands.feature:33]: I think I am missing something, but why are we removing that part of the test ?

File: features/apt_messages.feature (hunk 1 above)
Comment (orndorffgrant) [reply features/apt_messages.feature:122]: ah this was a quick hack to run apt-get upgrade without providing stdin and without actually upgrading.

I'll replace it by just using the --dry-run flag

File: features/staging_commands.feature (hunk 2 above)
Comment (orndorffgrant) [reply features/staging_commands.feature:33]: The new tests test the full process of creating the messages and the apt hook filling in these variables. So I didn't think we needed these anymore, which only test how the apt hook fills in variables and concatenates files.

But I can add them back if we think they add value!

Comment (lucasmoura) [reply features/staging_commands.feature:33]: Fair enough, just wanted to double check. No problem removing that

**Review by lucasmoura — APPROVED** [review]

File: features/apt_messages.feature [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+        Then stdout matches regexp:
+        """
+        Reading package lists...
+        Building dependency tree...
+        Reading state information...
+        Calculating upgrade...
+        The following security updates require Ubuntu Pro with 'esm-infra' enabled:
+          .*
+        Learn more about Ubuntu Pro for 16\.04 at https:\/\/ubuntu\.com\/16-04
+        0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded\.
```
Comment (orndorffgrant) [inline features/apt_messages.feature:119]:
```suggestion
        Reading package lists...
        Building dependency tree...
        Reading state information...
        Calculating upgrade...
        The following package was installed but can be removed:
          package1 package2...
        run autoremove
        The following security updates require Ubuntu Pro with 'esm-infra' enabled:
          package1 package2 package3 package4
        Learn more about Ubuntu Pro for 16\.04 at https:\/\/ubuntu\.com\/16-04
        The following packages will get upgraded:
          package1 package2
        0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded\.
```
@lechsandecki

**Pull request merged**